File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        
        int start = 0;
        int end = heights.length - 1;
        int h = 0;

        while (start < end){
            
            int temp = (end - start) * Math.min(heights[start],heights[end]);
            if (temp > h) h = temp;

            if (heights[start] < heights[end]){
                start++;
            } else {
                end--;
            }
        }
        return h;
    }
}

File: Data Structures & Algorithms/append-characters-to-string-to-make-subsequence/submission-0.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        i , j = 0, 0 

        while j < len(s) and i < len(t):
            if s[j] == t[i]:
                j += 1
                i += 1
                continue
            j += 1
        
        return len(t) - i